### PR TITLE
Use standard directories on Unix and Windows

### PR DESCRIPTION
**Pull request by thesbros**
Not sure what the coding style is, so if I made a mistake please tell me.

### CODE REVIEW

**Comment by thesbros** [comment]
Now that I think about it, the tests might need to be changed to account for this too.

**Comment by thesbros** [comment]
Fixed problems and rebased, tests should finish now.

File: index.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ function fail (err) {
 
 function openConfig (cb) {
   var userHome = require('user-home');
+  if (process.platform === 'linux' && userHome) {
+    userHome += '/.cache';
```
Comment (tkellen) [inline index.js]: I think you should scope this by if the platform is windows or not. OSX has a standard `~/.cache` path as well.  Also, if windows has a similar path, we should use it (and use `path.join` here as well).

Comment (thesbros) [inline index.js]: Changed.

**Comment by tkellen** [comment]
Before we merge something like this I just want to check with @sindresorhus to see if he wants to support this directly in the `user-home` module.

**Comment by thesbros** [comment]
That would probably be best fit for another module, isn't user-home only for getting the user home? It doesn't do anything else.

**Comment by tkellen** [comment]
Sure, but it could expose a second property? If @sindresorhus doesn't want
to add this maybe you could cut a module that builds off his and upstream
it here?

**Comment by sindresorhus** [comment]
I already have a module for this:
https://github.com/sindresorhus/xdg-basedir#cache

File: index.js [hunk 2]
```diff
@@ -27,6 +27,9 @@ function fail (err) {
 
 function openConfig (cb) {
   var userHome = require('user-home');
+  if (process.platform !== 'win32' && userHome) {
+    userHome = path.join(userHome, '.cache');
+  }
   var configpath = path.join(userHome || os.tmpdir(), configfile);
```
Comment (sindresorhus) [inline index.js]: I have a module for exactly this too: https://github.com/sindresorhus/home-or-tmp

**Comment by thesbros** [comment]
I feel like adding two modules just to userHome || os.tmpdir() and path.join(userHome, '.cache') is a bit overkill.

**Comment by sindresorhus** [comment]
Yeah, never mind the last suggestion, it's moot if you decide to use `xdg-basedir`.

**Comment by tkellen** [comment]
@thesbros I did some digging about Windows, the standard cache directory would be `path.join(userHome, 'AppData', 'Local')`. Since xdg-basedir doesn't take this into account, I'm :+1: to integrating it directly. Happy to merge once we have windows equally accounted for!

**Comment by thesbros** [comment]
Alright, done.

**Comment by tkellen** [comment]
thanks @thesbros!

**Comment by tkellen** [comment]
published as `2.0.6`

**Comment by tomxtobin** [comment]
Hmm, why was this reverted?

I finally tracked down what was creating the annoying ".v8flags.foo.bar.json" in my home directory to this package, saw that the issue was already solved by this PR in `2.0.6` (by nicely sticking it in the XDG cache dir) ... and then undone in `2.0.7`. I can't figure out why from reading the commits and issues.

**Comment by thesbros** [comment]
Seems it has to do with [this.](https://github.com/js-cli/js-v8flags/issues/22#issuecomment-112118432) Don't know why it was completely reverted rather than just making it create the folder.

**Comment by phated** [comment]
Yep, regressions abound were introduced.  It will probably be re-added when #29 is fixed.

**Comment by tkellen** [comment]
@thesbros it was completely reverted because I was in the middle of a consulting project when this change blew up. I'd love a PR that fixes the underlying problem so we can restore this functionality.

**Comment by Siilwyn** [comment]
Just went through the same hassle @tomxtobin, finally found the package doing this. ^^

So maybe #29 is fixed by #31 and then we could use standard directories! :raised_hands: 
Also I'd use the [env-paths](https://github.com/sindresorhus/env-paths) module instead of writing the logic.

**Comment by tkellen** [comment]
I'd happily take a PR for this if you're willing to be the front line for the inevitable fallout.

There are a fair few projects which depend on this:
https://www.npmjs.com/browse/depended/v8flags

**Comment by Siilwyn** [comment]
I think I would be willing...
Especially if a major version bump is allowed. :)

**Comment by phated** [comment]
No sindre modules. Sorry.

**Comment by Siilwyn** [comment]
@phated are you serious or just joking?

**Comment by phated** [comment]
Serious, I have a bunch of examples why not to but the basic rule is "don't"

**Comment by Siilwyn** [comment]
Hmm that's a bummer, I used this module several times and helped with fixing a bug so implementing the logic another time here seems in vain. Since the [whole module](https://github.com/sindresorhus/env-paths/blob/master/index.js) is only about 60 lines of code could you give it a chance/look?

**Comment by AlexTes** [comment]
@phated do you happen to have them written down somewhere?

**Comment by phated** [comment]
Locking this.  It's not the place to discuss this.

**Comment by tkellen** [comment]
@phated I think you're being a bit unreasonable here. If this functionality is something these folks want to add and they are willing to bear the responsibility of the impact, I see no reason to block them. @Siilwyn, please go ahead with your PR. If @phated is *that* hard-line about Sindre's modules, just inline the utility and the tests, the world will not explode because of the duplication.

**Comment by phated** [comment]
One of the reasons to not use his modules is that he is systematically dropping node 0.10 support from his modules (in some cases just to use an arrow function that makes no sense). All the gulp dependency tree is going to support node 0.10 for quite some time so we won't be depending on his modules. 

If you'd like to rewrite the code and give proper attribution to the original author or find a similar module, feel free to submit a PR.

**Pull request merged**